apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt

```diff
@@ -108,23 +113,28 @@ class RoomManager @Inject constructor(
     }
 
     override fun onConnectFailure(room: Room, twilioException: TwilioException) {
+        Timber.i("onConnectFailure -> room sid: %s", room.sid)
         this.localParticipantWrapper.participant = null
         handleError(ErrorResponse(twilioException.message ?: "Error", twilioException.explanation ?: "onConnectFailure"))
     }
 
     override fun onReconnecting(room: Room, twilioException: TwilioException) {
-
+        Timber.i("onReconnecting -> room sid: %s", room.sid)
     }
 
     override fun onReconnected(room: Room) {
-
+        Timber.i("onReconnected -> room sid: %s", room.sid)
     }
 
     override fun onDisconnected(room: Room, twilioException: TwilioException?) {
-
+        Timber.i("onDisconnected -> room sid: %s", room.sid)
+        if (!isDisconnectedFromUI) {
+            disconnect(false)
+        }
```

Comment:
I think we need to look for specific errors here to explicitly detect when a room is ended by the host. 

See this example: https://github.com/twilio/twilio-live-interactive-video/blob/6d01822ff59c6666754375472043b2270e40237b/apps/ios/LiveVideo/LiveVideo/Twilio/Room/RoomManager.swift#L86

Reply:
Thanks! This help me a lot to clarify how to handle multiple disconnection types.